src/backend/gl/src/command.rs

```diff
@@ -326,44 +327,50 @@ impl RawCommandBuffer {
 
     fn update_blend_targets(&mut self, blend_targets: &Vec<pso::ColorBlendDesc>) {
         let max_blend_slots = blend_targets.len();
+        if max_blend_slots == 0 {
+            return;
+        }
 
-        if max_blend_slots > 0 {
-            match self.cache.blend_targets {
-                Some(ref mut cached) => {
-                    if cached.len() < max_blend_slots {
-                        cached.resize(max_blend_slots, None);
-                    }
-                }
-                None => {
-                    self.cache.blend_targets = Some(vec![None; max_blend_slots]);
-                }
-            };
+        if self.cache.blend_targets.len() < max_blend_slots {
+            self.cache.blend_targets.resize(max_blend_slots, None);
         }
 
-        for (slot, blend_target) in blend_targets.iter().enumerate() {
+        let mut all_targets_same = true;
+        for blend_target in &blend_targets[1..] {
```

Comment:
should use iterators here, i.e. `let all_targets_same = blend_targets[1..].iter().all(|target| target == blend_targets[0])`

Reply:
Yeah I definitely should do that, wasn't thinking